spacegraph/spacegraph_codebase/trainer.py

```diff
@@ -240,6 +240,136 @@ def make_args_combine(args):
     return args_combine
 
 
+def make_enc_dec(args, pointset=None, feature_embedding= None):
```

Comment:
This is not new code; it's only an encapsulation of code that was previously part of the `__init__` of the `Trainer` class. In the new version, it is a separate function, and can be used in the `test.py` script.